java-checks/src/main/java/org/sonar/java/checks/regex/GraphemeClustersInClassesCheck.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.regex;
+
+import java.text.Normalizer;
+import java.util.Collections;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.java.regex.RegexParseResult;
+import org.sonar.java.regex.ast.CharacterClassTree;
+import org.sonar.java.regex.ast.CharacterClassUnionTree;
+import org.sonar.java.regex.ast.JavaCharacter;
+import org.sonar.java.regex.ast.PlainCharacterTree;
+import org.sonar.java.regex.ast.RegexBaseVisitor;
+import org.sonar.java.regex.ast.RegexTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+
+@Rule(key = "S5868")
+public class GraphemeClustersInClassesCheck extends AbstractRegexCheck {
+
+  private static final String MESSAGE = "Extract these Grapheme Clusters from the character class.";
```

Comment:
The rule description proposes a more precise message when the regex is "simple enough", however, I'm not sure to get why are the benefits. If you feel that it would be useful, let's discuss it!

Reply:
I think it would be valuable to use at least a singular/plural form if not using the simpler case.